operators/pkg/utils/test/integration.go

```diff
@@ -34,10 +47,39 @@ func RunWithK8s(m *testing.M, crdPath string) {
 
 	var err error
 	if Config, err = t.Start(); err != nil {
-		log.Error(err, "failed to start")
+		fmt.Println("failed to start test environment:", err.Error())
+		panic(err)
 	}
 
 	code := m.Run()
-	t.Stop()
+	if err := t.Stop(); err != nil {
+		fmt.Println("failed to stop test environment:", err.Error())
+	}
 	os.Exit(code)
 }
+
+// StarTestController sets up a manager and controller to perform reconciliations in background.
+// It must be stopped by calling the returned function.
+func StartTestController(t *testing.T, addToMgrFunc func(manager.Manager, operator.Parameters) error, parameters operator.Parameters) (k8s.Client, func()) {
```

Comment:
I would be tempted to call this function `StartManager` because starting controllers is the job of the `Manager` and `test` is already in the package name. Then we could remove the redundant comment `// start the test manager & controller` where this function is called.